Model/Filter/FilterBuilder.php

```diff
@@ -0,0 +1,196 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Model\Filter;
+
+use Magento\Catalog\Model\Layer\Filter\Item;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\Store;
+use Nosto\Operation\Recommendation\IncludeFilters;
+use Nosto\Operation\Recommendation\ExcludeFilters;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+
+class FilterBuilder
+{
+    /** @var IncludeFilters */
+    private $includeFilters;
+
+    /** @var ExcludeFilters */
+    private $excludeFilters;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var string */
+    private $brand;
+
+    /** @var NostoLogger */
+    private $logger;
+
+    /**
+     * FilterBuilder constructor.
+     * @param IncludeFilters $includeFilters
+     * @param ExcludeFilters $excludeFilters
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoLogger $logger
+     */
+    public function __construct(
+        IncludeFilters $includeFilters,
+        ExcludeFilters $excludeFilters,
+        NostoHelperData $nostoHelperData,
+        NostoLogger $logger
+    ) {
+        $this->includeFilters = $includeFilters;
+        $this->excludeFilters = $excludeFilters;
+        $this->nostoHelperData = $nostoHelperData;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     */
+    public function init(Store $store): void
+    {
+        $this->brand = $this->nostoHelperData->getBrandAttribute($store);
+    }
+
+    /**
+     * @param Item[] $filters
+     * @throws LocalizedException
+     */
+    public function buildFromSelectedFilters($filters)
+    {
+        foreach ($filters as $filter) {
+            $this->mapIncludeFilter($filter);
+        }
+    }
+
+    /**
+     * @param Item $item
+     * @throws LocalizedException
+     */
+    public function mapIncludeFilter(Item $item): void
+    {
+
+        $attributeModel = $item->getFilter()->getData('attribute_model');
+        if($attributeModel === null) {
+            return;
+        }
+
+        /** @var string $frontendInput */
+        $frontendInput = $attributeModel->getData('frontend_input');
+        if ($frontendInput === null) {
+            return;
+        }
+
+        $filterName = $item->getName();
+        $value = '';
+        switch ($frontendInput) {
+            case 'price':
+                $value = $item->getData('value');
+                break;
+            case 'select':
+            case 'multiselect':
+                $value = $item->getData('label');
+                break;
+            case 'date':
+                break;
+            case 'boolean':
+                $value = $item->getData('value') === '1';
```

Comment:
This looks flaky. If value is `1`(int) this will return false.
Cast to boolean, should manage those cases.